### ISSUE TITLE
Adjust theme for each account

### ISSUE THREAD

**Issue by hitam4450** [stefan-niedermann/nextcloud-deck#317]
Hi,

To make the Look & Feel compatible with the nextcloud app, Theming should be included.

I do have multiple instances of Nextcloud running and each instance does have a separate theming color to be able to distinguish between the instances right at looking at the screen !

I am sure, this will be very helpful for the NextCloud deck app as well !!

Best regards, hitam 4450

**Comment by stefan-niedermann** [comment]
@desperateCoder the `Account` class needs a property `themeColor` to achieve this. Just update it everytime you are reading the `Capabilities`-endpoint, this should be by far enough.

When you write an database update script, you can use `#0082c9`  as initial value.

PS.: I also thought about storing the server version for each account locally, we should talk about this topic before we do anything because you could do both database changes at once if you want.

@hitam4450 please don't expect quick progress since theming is a extremliy complicated topic (we need light fonts when the user has a dark `themeColor` and dark fonts with a light `themeColor`, additional the app supports a light and dark theme, so....

**Comment by desperateCoder** [comment]
For further development:

So we have to extract:
* the server-deck-version (default: 0.6.4)
* theming->color (default: `#0082c9`)
* theming->colorText (default: `#ffffff`)
* the maintenance-flag (default: false)

Capabilities calll when in maintenance mode:
```json
{
  "ocs": {
    "meta": {
      "status": "failure",
      "statuscode": 503,
      "message": "Service unavailable"
    },
    "data": []
  }
}
```
... and when not in maintenance:
```json
{
  "ocs": {
    "meta": {
      "status": "ok",
      "statuscode": 200,
      "message": "OK"
    },
    "data": {
      "version": {
        "major": 18,
        "minor": 0,
        "micro": 3,
        "string": "18.0.3",
        "edition": "",
        "extendedSupport": false
      },
      "capabilities": {
        "core": {
          "pollinterval": 60,
          "webdav-root": "remote.php/webdav"
        },
        "bruteforce": {
          "delay": 0
        },
        "files": {
          "bigfilechunking": true,
          "blacklisted_files": [
            ".htaccess"
          ],
          "directEditing": {
            "url": "https://my.fancy.domain.tld/ocs/v2.php/apps/files/api/v1/directEditing",
            "etag": "81d2d2d71d3931930a7f01c991ff84fc"
          },
          "versioning": true,
          "undelete": true
        },
        "activity": {
          "apiv2": [
            "filters",
            "filters-api",
            "previews",
            "rich-strings"
          ]
        },
        "ocm": {
          "enabled": true,
          "apiVersion": "1.0-proposal1",
          "endPoint": "https://my.fancy.domain.tld/index.php/ocm",
          "resourceTypes": [
            {
              "name": "file",
              "shareTypes": [
                "user",
                "group"
              ],
              "protocols": {
                "webdav": "/public.php/webdav/"
              }
            }
          ]
        },
        "deck": {
          "version": "0.8.0"
        },
        "dav": {
          "chunking": "1.0"
        },
        "files_sharing": {
          "api_enabled": true,
          "public": {
            "enabled": true,
            "password": {
              "enforced": false,
              "askForOptionalPassword": false
            },
            "expire_date": {
              "enabled": false
            },
            "multiple_links": true,
            "expire_date_internal": {
              "enabled": false
            },
            "send_mail": false,
            "upload": true,
            "upload_files_drop": true
          },
          "resharing": true,
          "user": {
            "send_mail": false,
            "expire_date": {
              "enabled": true
            }
          },
          "group_sharing": true,
          "group": {
            "enabled": true,
            "expire_date": {
              "enabled": true
            }
          },
          "default_permissions": 31,
          "federation": {
            "outgoing": true,
            "incoming": true,
            "expire_date": {
              "enabled": true
            }
          },
          "sharee": {
            "query_lookup_default": false
          },
          "sharebymail": {
            "enabled": true,
            "upload_files_drop": {
              "enabled": true
            },
            "password": {
              "enabled": true
            },
            "expire_date": {
              "enabled": true
            }
          }
        },
        "notifications": {
          "ocs-endpoints": [
            "list",
            "get",
            "delete",
            "delete-all",
            "icons",
            "rich-strings",
            "action-web"
          ],
          "push": [
            "devices",
            "object-data",
            "delete"
          ],
          "admin-notifications": [
            "ocs",
            "cli"
          ]
        },
        "password_policy": {
          "minLength": 8,
          "enforceNonCommonPassword": true,
          "enforceNumericCharacters": false,
          "enforceSpecialCharacters": false,
          "enforceUpperLowerCase": false,
          "api": {
            "generate": "https://my.fancy.domain.tld/ocs/v2.php/apps/password_policy/api/v1/generate",
            "validate": "https://my.fancy.domain.tld/ocs/v2.php/apps/password_policy/api/v1/validate"
          }
        },
        "spreed": {
          "features": [
            "audio",
            "video",
            "chat-v2",
            "guest-signaling",
            "empty-group-room",
            "guest-display-names",
            "multi-room-users",
            "favorites",
            "last-room-activity",
            "no-ping",
            "system-messages",
            "mention-flag",
            "in-call-flags",
            "notification-levels",
            "invite-groups-and-mails",
            "locked-one-to-one-rooms",
            "read-only-rooms",
            "chat-read-marker",
            "webinary-lobby",
            "start-call-flag",
            "chat-replies",
            "circles-support"
          ],
          "config": {
            "chat": {
              "max-length": 32000
            }
          }
        },
        "theming": {
          "name": "Niedermann IT",
          "url": "https://www.niedermann.it",
          "slogan": "a safe home for all your data",
          "color": "#007676",
          "color-text": "#ffffff",
          "color-element": "#007676",
          "logo": "https://my.fancy.domain.tld/index.php/apps/theming/image/logo?useSvg=1&v=24",
          "background": "https://my.fancy.domain.tld/core/img/background.png?v=24",
          "background-plain": false,
          "background-default": true,
          "logoheader": "https://my.fancy.domain.tld/index.php/apps/theming/image/logo?useSvg=1&v=24",
          "favicon": "https://my.fancy.domain.tld/index.php/apps/theming/image/logo?useSvg=1&v=24"
        }
      }
    }
  }
}

```

**Comment by desperateCoder** [comment]
@stefan-niedermann I'm kind of done here. Have fun with it!

**Comment by stefan-niedermann** [comment]
Will be released as a setting in version `0.7.0`.
